server.js

```diff
@@ -0,0 +1,82 @@
+var _ = require("lodash"),
+  bodyparser = require("body-parser"),
+  express = require("express"),
+  jsdom = require("jsdom"),
+  request = require("request"),
+  geocoderCn = require("./lib/geocoder.js"),
+  makeObjAndStringify = require("./lib/makeobjandstringify.js"),
+  piglatinify = require("./lib/piglatinify.js"),
+  randomJoke = require("./lib/jokes.js"),
+  randomString = require("./lib/strings.js");
+var app = express();
+var port = process.env.PORT || 3000;
+
+// Limit portions of app bringing in, so it doesn't load EVERYTHING
+app.use(bodyparser.json()); // Use the JSON portion of body parser
+app.use(bodyparser.urlencoded({ extended: true })); // Use url decoder
+app.use(express.static(__dirname + "/app/"));  // Links Our App Folder
+
+////////////////////// Endpoints/Routes //////////////////////////
+app.get("/", function(req, res) {
+  res.sendFile("index.html");     // We can send files from app folder
+  //res.send("hello, universe");
+});
+
+app.get("/jokes", function(req, res) {
+  res.json(randomJoke());
+});
+
+app.get("/strings", function(req, res) {
+  res.send(randomString());
+});
+
+// TODO: Hit Remote Server ONLY if haven't already loaded value before.
+app.get("/npmname", function(req, res) {
+  var siteAddress = "https://www.npmjs.com/package/wolverine";
+  request({ uri: siteAddress }, function(err, response, body) {
+    // Returns a JSON DOM Object; parses for the proper text
+    jsdom.env( {
+      html: (response["req"]["res"]["body"]),
+      scripts: [ "http://code.jquery.com/jquery-1.6.min.js" ],
+      done: function(err, window) {
+        var $ = window.jQuery;
+        var npmString = $("#npm-expansions").text();
+        var jsonObj = makeObjAndStringify(npmString);
+        res.send( jsonObj );
+        // var needObj = { text: npmString };
+        // res.send( JSON.stringify(needObj) );
+      }
+    });
+  });
+});
+
+// Posts are sent in the body
+app.post("/piglatin", function(req, res) {  // URL parser gives access to req
+  var firstname = piglatinify(req.body.firstname);
+  var lastname = piglatinify(req.body.lastname);
+  var piglatinated = { firstname: firstname, lastname: lastname };
+
+  res.json(piglatinated);  // Send object as JSON
+});
+
+// Geocoding
+app.post("/geocode", function(req, res) {
+  var fullAddress = geocoderCn(req.body.searchLocation);
+
+  // Hit Google Geocoding API Server
+  // Wanted to do in geocoder.js, but async cause return of undefined
+  // Never did figure out how to prevent an async return from module
```

Comment:
generally, if you need to prevent async and make something run "in order", you pass in a callback. The function does some work, then you call the callback - and that has your next "step" in it. But that can quickly lead to callback hell, so watch out. :)